src/system.h

```diff
@@ -278,55 +279,80 @@ struct System {
-                            // PNG.
+                            // Uhoh.. the decoder failed. Try to save the situation by skipping this image
                             anyimagesfailed = true;
-                            if (beforepng == wxInvalidOffset) return _(L"Cannot tell/seek document?");
-                            fis.SeekI(beforepng);
-                            // Now try to skip past this PNG
-                            uchar header[8];
-                            fis.Read(header, 8);
-                            uchar expected[] = {0x89, 'P', 'N', 'G', '\r', '\n', 0x1A, '\n'};
-                            if (memcmp(header, expected, 8)) return _(L"Corrupt PNG header.");
-                            dis.BigEndianOrdered(true);
-                            for (;;)  // Skip all chunks.
-                            {
-                                wxInt32 len = dis.Read32();
-                                char fourcc[4];
-                                fis.Read(fourcc, 4);
-                                fis.SeekI(len, wxFromCurrent);  // skip data
-                                dis.Read32();                   // skip CRC
-                                if (memcmp(fourcc, "IEND", 4) == 0) break;
+                            if (beforeimage == wxInvalidOffset) return _(L"Cannot tell/seek document?");
+                            fis.SeekI(beforeimage);
+                            switch(imt) {
+                                case wxBITMAP_TYPE_PNG: {
+                                    // Now try to skip past this PNG
+                                    uchar header[8];
+                                    fis.Read(header, 8);
+                                    uchar expected[] = {0x89, 'P', 'N', 'G', '\r', '\n', 0x1A, '\n'};
+                                    if (memcmp(header, expected, 8)) return _(L"Corrupt PNG header.");
+                                    dis.BigEndianOrdered(true);
+                                    for (;;)  // Skip all chunks.
+                                    {
+                                        wxInt32 len = dis.Read32();
+                                        char fourcc[4];
+                                        fis.Read(fourcc, 4);
+                                        fis.SeekI(len, wxFromCurrent);  // skip data
+                                        dis.Read32();                   // skip CRC
+                                        if (memcmp(fourcc, "IEND", 4) == 0) break;
+                                    }
+                                    break;
+                                }
+                                case wxBITMAP_TYPE_JPEG: {
+                                    // Now try to skip past this JPEG
+                                    dis.BigEndianOrdered(true);
+                                    for (;;)  // Skip all chunks.
+                                    {
+                                        wxInt32 len = dis.Read32();

```

Comment:
This code is identical to the PNG skipping code.. I doubt it skips a JPEG correctly. This code must skip past a JPEG correctly, or if you don't know how to do that, for now you could instead return an error saying "could not load file because of corrupted embedded jpeg" or similar, but of course that means if it happens to people they will have lost their file until this functionality is implemented.

Reply:
Ok, I reduced the code to a return with the error message as I do not know whether the End of image sequence may also appear before.